### DISCUSSION TITLE
如何解决 PDF 预览的 CORS 问题？本地存储

### DISCUSSION THREAD

**Discussion by TTtrace** [AlistGo/alist#4072] [Q&A]
我用的是本地存储，想实现pdf 预览。看了下文档说要求 https + cors 解决才可以。我申请了 ssl 证书，https 访问没问题了。不过 CORS 的问题该如何解决呢？文旦里说的是需要网盘支持 CORS，可没说本地存储该怎么解决。

我尝试了设定了 site_url 为自己的网站域名（带https），也试了在 nginx 反向代理中添加 X-Forwarded-Proto 头，都还是没用。

错误如下：

![image](https://user-images.githubusercontent.com/39454079/230571133-de431937-dd53-49c7-8517-fa9724cbe622.png)

![image](https://user-images.githubusercontent.com/39454079/230571317-3b699d8b-de86-4c37-b8ec-adc6d85e4e8b.png)

**Comment by VirtualCoder0** [comment]
同问

**Reply by Rektyfikowany** [reply]
本地代理一般不会遇到CORS问题，[server/router.go](https://github.com/alist-org/alist/blob/main/server/router.go#L143)默认允许跨域，F12看下请求`你的域名/p/挂载路径`的响应头是否包含`Access-Control-Allow-Origin`，不含则需检查nginx配置

**Comment by kukudelang** [comment]
浏览器跨域访问就会出这个问题，我用手机termux启动alist, 本机浏览器访问没问题，局域网内其他手机访问该手机的alist就会出现你一样的问题，我查了下csdn，里面说这个是PDF.JS的问题，需要编译这个JS文件允许跨域访问或者你使用反向代理
